samples/theia-hello-world-frontend-plugin.devfile

```diff
@@ -0,0 +1,65 @@
+specVersion: 0.0.1
+name: theia-hello-world-plugin
+projects:
+  - name: che-theia-samples
+    source:
+      type: git
+      location: 'https://github.com/eclipse/che-theia-samples.git'
+tools:
+  - name: theia-editor
+    type: cheEditor
+    id: org.eclipse.che.editor.theia:1.0.0
+  - name: exec-plugin
+    type: chePlugin
+    id: che-machine-exec-plugin:0.0.1
+  - name: theia-dev
+    type: kubernetes
+    localContent: |
```

Comment:
Please check that it is a working example. I assume that validator will require local field https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-devfile/src/main/resources/schema/devfile.json#L113

Reply:
Fixed. Still unclear for me why the hell we need this field here...

Reply:
I would avoid that syntax in the examples. This is hard to write (you got it wrong the first time and I am sure users will do the same mistake) and hard to read.

Instead we should move the content in a distinct yaml file and add a reference to the raw URL. If it's not supported yet you can either remove this example or already use the url syntax that should be implemented in a few days.

Reply:
I am not so optimistic about `url`  (i see even the dockerimage still not applied on prod, which is merged a long time ago). I think it's not less than month, during which example will be broken. So lets leave it or remove at all.

Reply:
> Note that Lukas will merge his PR soon, then it will be possible to accept a factory with devfile raw content link even if it references local files eclipse/che#12800.
@metlos Could you confirm it?

As long as the local references are stored on the same host and accessible on the same path relative to the devfile location as their `local` name in the devfile suggests then yes, it should be possible.

E.g. the following will work when sending a link to the devfile location to the factory:
Devfile:
```yaml
...
  - name: ourTool
    type: kubernetes
    local: ../deployments/our-tool.yaml
...
```

Devfile location: `http://acme.com/metadata/foo/ide-config/devfile.yaml`
Local location: `http://acme.com/metadata/foo/deployments/our-tool.yaml`